### PR TITLE
When your test scripts fail on a Wednesday...

### PR DESCRIPTION
**Pull request by johnsyweb**
# Context

I've allowed our fork to get behind and I want to correct this. When I started to look around to see what had changed since my last change, I made some refactoring (for a future pull request). I went to bed and then next morning my tests were failing.

# Change

`bin/late_afternoon_seal.sh` will now run without error on Wednesdays!

# Considerations

The hard-coded values in `#check_if_quotes` feel like they should be configuration items. For a future PR, I guess.

### CODE REVIEW

File: lib/slack_poster.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@ def initialize(webhook_url, team_channel, mood)
     @webhook_url = webhook_url
     @team_channel = team_channel
     @mood = mood
+    @today = Date.today
+    @postable_day = !today.saturday? && !today.sunday?
```
Comment (johnsyweb) [inline lib/slack_poster.rb]: Days of working time in Israel are Sunday through Thursday (for a five-day workweek). Perhaps this should be configurable, too?

**Comment by johnsyweb** [comment]
@binaryberry I'm interested in your feedback here, I've a few refactorings that I'd like to send your way, too :)

**Comment by binaryberry** [comment]
Sure @johnsyweb, sorry it's taking me a while to look at this properly, I've a lot going on. Hoping to have a look this weekend!

**Comment by binaryberry** [comment]
And as always, thanks for the contributions and I look forward to learning from them :-)

File: lib/slack_poster.rb (hunk 1 above)
Comment (binaryberry) [inline lib/slack_poster.rb]: That would be a good idea! A "non-work days" attribute could be added to the team config?

File: lib/slack_poster.rb [hunk 2]
```diff
@@ -19,15 +21,18 @@ def create_poster
 
   def send_request(message)
     if ENV['DRY']
+      puts "Will#{' not' unless postable_day} post #{mood} message to #{channel} on #{today.strftime('%A')}"
```
Comment (binaryberry) [inline lib/slack_poster.rb]: Nice. I've had issues testing it on weekend days so this is very helpful.

File: lib/slack_poster.rb (hunk 1 above)
Comment (johnsyweb) [inline lib/slack_poster.rb]: I'd probably prefer positive predicates (such aliteration!) and configure _work_ days, but yes!

File: lib/slack_poster.rb [hunk 3]
```diff
@@ -53,14 +58,14 @@ def assign_poster_settings
     elsif @mood == "angry"
       @mood_hash[:icon_emoji]= ":#{@season_symbol}angrier_seal:"
       @mood_hash[:username]= "#{@season_name}Angry Seal"
-    elsif @mood == "tea" && @postable_day
+    elsif @mood == "tea"
       @mood_hash[:icon_emoji]= ":manatea:"
       @mood_hash[:username]= "Tea Seal"
-    elsif @mood == "charter" && @postable_day
+    elsif @mood == "charter"
```
Comment (binaryberry) [inline lib/slack_poster.rb]: oh, yes. Adding the condition here too was overkill. Thanks!

File: spec/slack_poster_spec.rb [hunk 4]
```diff
@@ -67,4 +67,36 @@
...
+      it 'is not time for tea' do
+        expect(fake_slack_poster).not_to receive(:send_message)
+        slack_poster.send_request('earl grey')
+      end
+    end
+
+    context 'on Friday' do
+      let(:friday) { Time.local(2016, 4, 15) }
+      before { Timecop.freeze(friday) }
+      it 'is tea time' do
```
Comment (binaryberry) [inline spec/slack_poster_spec.rb]: 👍 🍵 😂

**Comment by binaryberry** [comment]
@johnsyweb Another great contribution! Thanks for your attention to detail and useful additions.
Sorry it took me ages to review. If you rebase (to take into account #83 ), I'll happily merge it.

File: lib/slack_poster.rb (hunk 1 above)
Comment (binaryberry) [inline lib/slack_poster.rb]: Sounds good! (which does sound better than "sounds non-bad")

**Comment by johnsyweb** [comment]
Happy to help. Looks like this merges cleanly with `master`.

**Pull request merged**